src/cores/ads/Ad.jsx

```diff
@@ -1,32 +1,34 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
-import { Link } from 'react-router-dom';
+import React from "react";
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
 
 const Ad = ({ id, isVisible, showModal }) => {
-  const ad = useSelector(state => state.ads.ads.find(ad => ad.id === id));
-  
+  const ad = useSelector((state) => state.ads.ads.find((ad) => ad.id === id));
```

Comment:
Podrian crear un selector en algun otro archivo e importarlo aqui, recomiendo tambien usar re-select de redux-toolkit

```suggestion
  const ad = useSelector(selectAdById(id));
```

(Esto es mega-opcional, pero ayuda a mejorar la legibilidad del codigo)